### PR TITLE
add strict mode to zip code rules

### PR DESCRIPTION
**Pull request by 7nohe**
```
$rules = ['zip_code' => 'zip_code:strict'];
$input1 = ['zip_code' => '1234567'];
$input2 = ['zip_code' => '123-4567'];
Validator::make($input1, $rules)->passes(); // false
Validator::make($input2, $rules)->passes(); // true
```

### CODE REVIEW

File: src/Rules/ZipCode.php [hunk 1]
```diff
@@ -6,6 +6,13 @@
 
 class ZipCode implements Rule
 {
+    public $parameters;
```
Comment (kozo) [inline src/Rules/ZipCode.php]: Set the default value.

```suggestion
    public $parameters = [
        'strict' => false
    ];
```

File: src/Rules/ZipCode.php [hunk 2]
```diff
@@ -18,6 +25,11 @@ class ZipCode implements Rule
     public function passes($attribute, $value)
     {
         $regex = '/^\d{3}\-?\d{4}$/';
+
+        // in strict mode, $value must include a hyphen
+        if ($this->parameters[0] == 'strict') {
```
Comment (kozo) [inline src/Rules/ZipCode.php]:
```suggestion
        if ($this->parameters['strict'] === true) {
```

File: src/Validators/ValiiValidator.php [hunk 3]
```diff
@@ -99,6 +99,12 @@ public function validateZenkaku(string $attribute, $value, array $parameters): b
     public function validateZipCode(string $attribute, $value, array $parameters): bool
     {
         $regex = '/^\d{3}\-?\d{4}$/';
+
+        // in strict mode, $value must include a hyphen
+        if ($parameters[0] == 'strict') {
```
Comment (kozo) [inline src/Validators/ValiiValidator.php]:
```suggestion
        if ($this->parameters['strict'] === true) {
```

File: tests/Unit/Rules/ZipCodeTest.php [hunk 4]
```diff
@@ -34,12 +34,35 @@ public function test_郵便番号チェック($zipCode, $expect)
...
+     *
+     * @dataProvider providerZipCodeWithHyphen
+     * @param $tel string テストデータ
+     * @param mixed $expect
+     */
+    public function test_郵便番号チェック（ハイフンあり）($zipCode, $expect)
+    {
+        $rule = [
+            'name' => [
+                new ZipCode(['strict'])
```
Comment (kozo) [inline tests/Unit/Rules/ZipCodeTest.php]:
```suggestion
                new ZipCode(['strict' => true])
```

File: src/Rules/ZipCode.php [hunk 5]
```diff
@@ -18,6 +25,11 @@ class ZipCode implements Rule
...
+          $regex = '/^\d{3}\-\d{4}$/';
```
Comment (kozo) [inline src/Rules/ZipCode.php]: Please change to 4 spaces indent.

**Review by kozo — COMMENTED** [review]
thank you.

**Review by KobaruTakahiro — APPROVED** [review]
thank you. LGTM!

**Pull request merged**